scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -385,14 +401,11 @@ public boolean isApproximation() {
         //localDerivative has been computed with what is now xn_m1 and xn_m2
         x_n_double = x_n_double - (ff_x_nminus1.in(Duration.MICROSECONDS) / localDerivative);
         x_n = Duration.of((long) x_n_double, Duration.MICROSECONDS);
-        //outside of domain, diverging
-        if (x_n_double < dxLow_long || x_n_double > dxHigh_long) {
-          throw new DivergenceException("Looking for root out of prescribed domain :[" + dxLow + "," + dxHigh + "]",
-                                        new History<>(history));
-        }
         ff_x_n = ff.valueAt(x_n);
-        history.put(x_n, ff_x_n);
-        if (ff_x_n.noShorterThan(Duration.negate(toleranceYLow)) && ff_x_n.noLongerThan(toleranceYHigh)) {
+        history.add(Pair.of(x_n, ff_x_n));
+        if (ff_x_n.noShorterThan(Duration.negate(toleranceYLow)) &&
+            ff_x_n.noLongerThan(toleranceYHigh) &&
+            (x_n_double >= dxLow_long && x_n_double <= dxHigh_long)){
```

Comment:
However, the final solution needs to be in the given bounds which is why this check is added here.

Reply:
This seems like another good comment to leave in-code too :)